src/harness/sas_testcase.py

```diff
@@ -282,6 +282,49 @@ def assertChannelsContainFrequencyRange(self, channels, frequency_range):
     self.assertEqual(channels[0]['frequencyRange']['highFrequency'],
                      frequency_range['highFrequency'])
 
+  def assertChannelsOverlapFrequencyRange(self, channels, frequency_range,
```

Comment:
Please add some unit tests, similar to what was done for the other function, so that we can be sure this works as intended? I think there are a number of bugs below and they would be quickly caught if we had some unit tests.

Reply:
will incorporate unit test in.

Reply:
Thanks! To save some time and effort, I'll wait to see the unit tests before reviewing this function further.

To save *yourself* some time and effort, I think you can mostly use the original unit tests and edit as needed.

Then to de-duplicate the code a bit, we can use
```
assertChannelsOverlapFrequencyRange(..., constrain_low=True, constrain_high=True)
```
instead of
```
assertChannelsContainFrequencyRange(...)
```
in SIQ_3 and remove assertChannelsContainFrequencyRange().

Reply:
Test cases added and passed all.